docs/developer-guide.md

```diff
@@ -0,0 +1,42 @@
+# Developer guide
+
+Have a look at the [user guide](user-guide.md) to familiarize yourself with things like rule naming conventions.
+
+## Running tests
+
+You can run the tests via:
+
+```console
+npm test
+```
```

Comment:
Let's add a line here about running single tests, something like this (I know if I were coming into this codebase I would at first be confused about this):

To run tests in a single file, instead of all the tests at once, you'll need to use `babel-tape-runner` (because the codebase is ES6). For example,

```console
babel-tape-runner src/rules/media-feature-name-no-vendor-prefix/__tests__/index.js
```